kernel/tls_initializer/src/lib.rs

```diff
@@ -156,7 +160,7 @@ impl TlsInitializer {
         #[cfg(target_arch = "aarch64")]
         let virt_addr_value = original_offset;
 
-        tls_section.virt_addr = VirtualAddress::new(virt_addr_value).ok_or(())?;
+        tls_section.virt_addr = VirtualAddress::new(virt_addr_value).ok_or(TlsInitializerError::InvalidData)?;
```

Comment:
this should use an error variant related to the TLS section having an invalid virtual address.